snap/snapcraft.yaml

```diff
@@ -1,12 +1,87 @@
 name: dcgm
 base: core24
 adopt-info: dcgm
-summary: Snap for NVIDIA DCGM and DCGM exporter
+summary: Snap for NVIDIA DCGM and DCGM-Exporter
 license: Apache-2.0
 contact: solutions-engineering@lists.canonical.com
 description: |
-  This snap includes NVIDIA DCGM and DCGM exporter to provide GPU monitoring via
-  Prometheus metrics. This snap does not include the dcgmproftester.
+  This snap includes NVIDIA DCGM and DCGM-Exporter to manage and monitor NVIDIA GPUs via the CLI or via Prometheus metrics.
+  For instance, the snap can be used to collect the metrics and make Grafana dashboard for data visualization.
+
+  The snap includes the following components:
+  - DCGM: Data Center GPU Manager
+  - DCGM-Exporter: a Prometheus exporter for DCGM metrics
+
+  Please see the links at the bottom of the page for more details about the included components and their purpose.
+
+  **How-To**
+  ---
+
+  **Install the snap**
+
+  ```
+  sudo snap install dcgm
+  ```
+
+  **How to enable metrics collection:**
+
+  ```
+  # Start the DCGM-Exporter service
+  sudo snap start dcgm.dcgm-exporter
+
+  # Get the metrics
+  curl -s localhost:9400/metrics
+  ```
+
+  **Note**: The `DCGM-Exporter` service is disabled by default. If you wish to collect metrics for monitoring,
+  see how to enable the exporter in the section above.
+
+  **How to configure the snap services**
+
+  The DCGM snap provides several configuration options that can be customized through the `snap` CLI.
+  For example:
+
+  ```
+  # Get all the configuration options
+  sudo snap get dcgm
+
+  # Set the NV-Hostengine port
+  sudo snap set dcgm nv-hostengine-port=5577
+
+  # Restart the NV-Hostengine service to apply the changes
+  sudo snap restart dcgm.nv-hostengine
+  ```
+
+  **Reference**
+  ---
+
+  Configurations available:
+
+  - `nv-hostengine-port`: The port on which the NV-Hostengine listens.
+    The default is `5555`.
+  - `dcgm-exporter-address`: The bind address which the DCGM-Exporter exposes for the metrics.
+    The default is `:9400`.
+  - `dcgm-exporter-metrics-file`: The name of the custom CSV metrics file can be provided (only the name, not the path).
+    The file should be placed in the `/var/snap/dcgm/common/` directory.
+    The default metrics are located in `/snap/dcgm/current/etc/dcgm-exporter/default-counters.csv`.
+    Please refer to the DCGM-Exporter repository link at the bottom of the page for more information on the CSV file format.
```

Comment:
```suggestion
    Please refer to the [DCGM-Exporter repository](https://github.com/NVIDIA/dcgm-exporter?tab=readme-ov-file#changing-metrics) for more information on the custom metrics file format.
```

...since CSV is a well defined file format

Reply:
The reason why I don't put links along the actual text is that the snapshot does not support the `[]()` markdown, so the links will pollute the text. That is why every links are at the bottom

Reply:
I'm not sure of the convention on the snap store pages, but a common way to reference particular urls with polluting the text is to use footnote-style references:

```suggestion
    Please refer to the DCGM-Exporter repository[0] for more information on the CSV file format.
    
    ...
    
    [0]: https://github.com/NVIDIA/dcgm-exporter?tab=readme-ov-file#changing-metrics
```

This is usually easier to follow and scales to many links.

Reply:
I could, but if you feel like this is the only place where this can be used, at the beginning of the description, there is another reference to the links, but I can't refer to the particular one(I refer to at least two of them). But it is a good approach. I'll take a note. Thanks!

Reply:
Please note that snapcraft does not support real "markdown", the following are shown in the [listing page](https://snapcraft.io/dcgm/listing): 
```
Show supported markdown syntax

Bold: **Foo**

URLs: https://foo.bar

Lists: * Foo

Italics: _Foo_

Code: Text indented with 3 spaces of inside `
```

The best way to check the final result is put this description into the listing page, and click preview (snapcraft  preview appears to be broken, at the time of writing)

Reply:
Yeah, It's broken for me, I know that it doesn't support the real "markdown", but I think I only used the supported syntax

Reply:
Yeah, so I am not sure if `[]()` or `xxx[]` syntax will work or not

Reply:
`[]()` syntax will not be interpreted as a link by snap store; it will be displayed as is (as far as I know).

`xxx[]` syntax will also not be processed, but it's a long standing plain text annotation for footnotes.

Reply:
👍🏼 I thought you meant creating a link

Reply:
Yep sorry I realise that was a bit ambiguous...